common/runtime.hpp

```diff
@@ -31,6 +31,10 @@ namespace ILLIXR {
 
 	};
 
+#ifdef ILLIXR_MONADO_MAINLINE
+	extern "C" runtime* runtime_factory();
+#else
 	extern "C" runtime* runtime_factory(GLXContext appGLCtx);
+#endif /// ILLIXR_MONADO_MAINLINE
```

Comment:
New `ILLIXR_MONADO_MAINLINE` build flag to support Monado integrated and mainline vesions on the same branch. After merging to `dev-monado`, `dev-monado` can be merged into `master`.

Reply:
> New `ILLIXR_MONADO_MAINLINE` build flag to support Monado integrated and mainline vesions on the same branch. After merging to `dev-monado`, `dev-monado` can be merged into `master`.

Switched base branch to `master` to bypass the Resolve Conflicts step before merging.

Reply:
Just have one overload of `runtime_factory` and pass `nullptr` if you have no `GLXContext`.

Reply:
Oh, is this seeking compatibility with the fork of ILLIXR maintained inside Monado?

Reply:
Yep! Maintaining another fork/branch is a bit cumbersome, so now we can compile both from the same branch using the `monado` action. Luckily the number of places where the runtime API is different is small :relieved:.

> Just have one overload of runtime_factory and pass nullptr if you have no GLXContext.

I am not sure, but I think since this is a C-style `extern`-linked function, we can't do function overloading.
[SO post](https://stackoverflow.com/questions/4866043/overload-resolution-with-extern-c-linkage)

Reply:
Another way to implement this is to guard only the arguments, as I've done for the `runtime_impl` constructor:
```cpp
extern "C" runtime* runtime_factory(
#ifdef ILLIXR_MONADO_MAINLINE
    GLXContext appGLCtx
#endif /// ILLIXR_MONADO_MAINLINE
);
```

Reply:
Ah, ok. I see why it is this way now.